### PR TITLE
add XML XSD version checking

### PR DESCRIPTION
**Pull request by dom54**

### CODE REVIEW

File: src/main/java/com/whiteclarkegroup/liquibaselinter/config/rules/core/AllowableXsdVersionsRule.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    @Override
+    public boolean invalid(DatabaseChangeLog changeLog) {
+        String xsdVersion = getXsdVersion(changeLog);
+        return !ruleConfig.getValues().contains(xsdVersion);
+    }
+
+    private String getXsdVersion(DatabaseChangeLog changeLog) {
+        try (FileReader reader = new FileReader(changeLog.getPhysicalFilePath());){
+            XMLEventReader eventReader = factory.createXMLEventReader(reader);
```
Comment (davidjgoss) [inline src/main/java/com/whiteclarkegroup/liquibaselinter/config/rules/core/AllowableXsdVersionsRule.java:51]: Wondering if we could use existing extension hooks to grab the file content? Even if it means a little refactoring

https://liquibase.jira.com/wiki/spaces/CONTRIB/pages/2424849/ChangeLogParser

**Comment by davidjgoss** [comment]
Thanks for doing this!

Could you also add documentation for the new rule:
- New file under https://github.com/whiteclarkegroup/liquibase-linter/tree/master/docs/rules
- Add to sidebar array in https://github.com/whiteclarkegroup/liquibase-linter/blob/master/website/sidebars.json

File: src/main/java/com/whiteclarkegroup/liquibaselinter/config/rules/core/AllowableXsdVersionsRule.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    public void configure(RuleConfig ruleConfig) {
+        super.configure(ruleConfig);
+        if (ruleConfig.hasPattern()) {
+            this.dbChangelogPattern = Pattern.compile(ruleConfig.getPatternString());
+        }
+    }
+
+    @Override
+    public boolean invalid(DatabaseChangeLog changeLog) {
+        String xsdVersion = getXsdVersion(changeLog);
```
[the remark this answers is not in the document]
Comment (dom54) [reply src/main/java/com/whiteclarkegroup/liquibaselinter/config/rules/core/AllowableXsdVersionsRule.java:45]: ah good point

**Comment by dom54** [comment]
Considering whether "liquibase.prefer.internal.xsd" in later versions of liquibase would be a better bet

**Comment by davidjgoss** [comment]
@dom54 ah didn't know about that. The [commit](https://github.com/liquibase/liquibase/pull/754/commits/9490dc1acbcbca1702d5fb65c6c2d97e653fe196) looks like it would have been in 3.6.1, maybe we can test that internally a bit.

**Comment by davidjgoss** [comment]
@dom54 do you think this is redundant now because of `liquibase.prefer.internal.xsd`

**Comment by dom54** [comment]
Yes, we can kill this one

**Comment by dom54** [comment]
Closing as later versions of liquibase prevent this being an issue ( I'll reopen if it proves not to be the case )

**Pull request closed without merging**